src/Http/Request.php

```diff
@@ -115,4 +117,24 @@ public function getPsrRequest(): RequestInterface
     {
         return $this->psrRequest;
     }
+
+    public function getParseCallback(): Closure
+    {
+        return $this->parseCallback;
+    }
+
+    public function setParseCallback(Closure $parseCallback): void
+    {
+        $this->parseCallback = $parseCallback;
+    }
+
+    public function getResponse(): ?Response
+    {
+        return $this->response;
+    }
+
+    public function setResponse(?Response $response): void
+    {
+        $this->response = $response;
+    }
```

Comment:
The request class is supposed to be immutable. I know it's a little inconsistent in the current implementation but I aim to fix this for the next major version.

So instead of overriding the current field, it should return a new `Request` with the provided response set. The method should then also be renamed to something like `withResponse` since we're no longer setting anything.

```suggestion
    public function withResponse(?Response $response): self
    {
        $clone = clone $this;
        $clone->response = $response;
        
        return $clone
    }
```